### PR TITLE
Slightly improve histogram feature

### PR DESCRIPTION
**Pull request by aswaterman**
In creating #1191, I noticed the histogram prints in order of PC, rather than order of frequency.  This isn't terribly useful, so swap that around.  After doing so, we can switch to `unordered_map` without a functional change, for a significant speedup.

### CODE REVIEW

**Review by scottj97 — APPROVED** [review]
👍

**Review by jerryz123 — APPROVED** [review]

**Review by scottj97 — CHANGES_REQUESTED** [review]
Please hold off on merging. I've hit some kind of incremental build issue that might be related to this PR. I'm investigating.

**Comment by aswaterman** [comment]
I could imagine the churn related to the instruction template changes in #1189 and #1190 might be to blame. Please merge if you determine this one is innocent.

**Comment by scottj97** [comment]
Maybe you can make some sense of what I've found.

| Clean build first on:  | Then incremental build on: | Outcome |
| ---------------- | --------------------- | ------- |
| #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | #1190 (edcf2d59b7019798f5c2ab303734883936c0b947) | ok      |
| #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | #1191 (b5f68dc55c2b310d24db35a3c7df1ad86509ab61) | ok      |
| #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | #1192 (d6e5a0ed89f1eb918f7a01e9fc7cdf7e71d6d791) | BROKEN! |
| #1190 (edcf2d59b7019798f5c2ab303734883936c0b947) | #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | ok      |
| #1190 (edcf2d59b7019798f5c2ab303734883936c0b947) | #1191 (b5f68dc55c2b310d24db35a3c7df1ad86509ab61) | ok      |
| #1190 (edcf2d59b7019798f5c2ab303734883936c0b947) | #1192 (d6e5a0ed89f1eb918f7a01e9fc7cdf7e71d6d791) | ok      |
| #1191 (b5f68dc55c2b310d24db35a3c7df1ad86509ab61) | #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | ok      |
| #1191 (b5f68dc55c2b310d24db35a3c7df1ad86509ab61) | #1192 (d6e5a0ed89f1eb918f7a01e9fc7cdf7e71d6d791) | ok      |
| #1192 (d6e5a0ed89f1eb918f7a01e9fc7cdf7e71d6d791) | #1189 (3cb3d10f058a9ceec6ec93f2aba4c49e2eaa58c1) | BROKEN! |
| #1192 (d6e5a0ed89f1eb918f7a01e9fc7cdf7e71d6d791) | #1190 (edcf2d59b7019798f5c2ab303734883936c0b947) | ok      |

"Broken" means that `vsetvli` (and possibly any vector instruction?) core dumps.

I haven't tried going to/from any older points before #1189 was merged.

**Comment by scottj97** [comment]
Another interesting data point: starting with a clean build from #1189, then incremental to #1190, then incremental to #1192, it's broken only once I reach #1192.

**Comment by aswaterman** [comment]
I can't make sense of it from that data.  I'll try to replicate your results so I can play around with the problem.

**Comment by aswaterman** [comment]
OK, at least I can confirm that I can replicate it.  I'll dig into it after my next meeting.

**Comment by aswaterman** [comment]
@scottj97 Figured it out.  It's a bit esoteric.  We're using `ar rcs` to build the archives prior to linking, which preserves and adds to the archive if it already existed.  This usually works by coincidence because our object files don't change names frequently, so they'll still overwrite their predecessors within the archive.  But because #1189 made new names for some of the objects, we now might end up with multiple objects in the archive with functions of the same name.

In the event of multiple objects satisfying a dependence at link time, I'm not sure which one gets chosen (alphabetical? order added to archive? arbitrary?) but the point is, the wrong version of the function is getting linked in.  From there, it's obvious how the result might be a segfault.

The fix is to delete the archive before building it.  Unfortunately, we're stuck with this broken intermediate state in our tree, but so be it.

I'll PR that fix before merging any other PRs.

**Comment by scottj97** [comment]
> frequently, so they'll still overwrite their predecessors within the archive. But because #1189 made new names for some of the objects, we now might end up with multiple objects in the archive with functions of the same name.

But #1190 changes the file names back to the same as before, yes?

If so, as long as I always include both #1189 and #1190, or neither #1189 nor #1190, then I shouldn't have an issue?

**Comment by aswaterman** [comment]
@scottj97 Right, that passes the smell test.

**Review by scottj97 — APPROVED** [review]
Incremental build issue solved and fixed in #1193

**Pull request merged**
